ng/manage.ts

```diff
@@ -1608,78 +1608,82 @@ angular.module("armExplorer", ["ngRoute", "ngAnimate", "ngSanitize", "ui.bootstr
-    var resourceInfo = GetResourceTypeAndName(value);
+    var resourceTypeAndName = getResourceTypeAndName(value);
+    // We need this variation to avoid passing in duplicate
+    // -ResourceName into New-AzureRmResource
+    var resourceType = getResourceTypeAndName(value, true);
     var children = value.resourceDefinition.children;
 
     // add GET related cmdlet if available
     var getCmdlet = "Get-AzureRmResource ";
     var apiVersion = value.resourceDefinition.apiVersion;
     var isCollection = "";
     if (value.httpMethod.toLowerCase().indexOf("get") != -1) {
-        returnString += "# GET " + GetActionName(value.url) + "\n";
+        returnString += "# GET " + getActionName(value.url) + "\n";
         if (typeof children === "string" && !Array.isArray(children) && value.url.split("/").length > 7) {
             isCollection = " -isCollection";
         }
-        returnString += getCmdlet + resourceInfo + isCollection + " -ApiVersion " + apiVersion + "\n\n";
+        returnString += getCmdlet + resourceTypeAndName + isCollection + " -ApiVersion " + apiVersion + "\n\n";
     }
     else if (value.httpMethod.toLowerCase().indexOf("post") != -1 && value.url.indexOf("list") != -1) {
-        returnString += "# List " + GetActionName(value.url.replace("/list", "")) + "\n";
-        returnString += "$resource = Invoke-AzureRmResourceAction " + resourceInfo + " -Action list -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n";
+        returnString += "# List " + getActionName(value.url.replace("/list", "")) + "\n";
+        returnString += "$resource = Invoke-AzureRmResourceAction " + resourceTypeAndName + " -Action list -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n";
         returnString += "$resource.Properties\n\n";
     }
 
     // add SET related cmdlet if available
     if (value.resourceDefinition.actions.some(a => (a === "PATCH" || a === "PUT"))) {
-        returnString += "# SET " + GetActionName(value.url) + "\n";
+        returnString += "# SET " + getActionName(value.url) + "\n";
         returnString += "$PropertiesObject = @{\n\t#Property = value;\n}\n";
 
         // handle secure GET
         if (value.resourceDefinition.actions.includes("GET")) {
-            returnString += "Set-AzureRmResource -PropertyObject $PropertiesObject " + resourceInfo + " -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n\n";
+            returnString += "Set-AzureRmResource -PropertyObject $PropertiesObject " + resourceTypeAndName + " -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n\n";
         }
         else {
-            returnString += "New-AzureRmResource -PropertyObject $PropertiesObject " + resourceInfo + " -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n\n";
+            returnString += "New-AzureRmResource -PropertyObject $PropertiesObject " + resourceType + " -ApiVersion " + value.resourceDefinition.apiVersion + " -Force\n\n";
```

Comment:
No matter which resource i try i can never reach that code path ^. Any hints on how to test this one?